### PR TITLE
Enable PlanResourceChange

### PR DESCRIPTION
**Pull request by t0yv0**
Enable PlanResourceChange by default for every resource in the provider. This pulumi-terraform-bridge feature was incubated under a flag and deployed selectively or quite some time. It should be ready to become the new default. Improvements include prevention of panics and undesirable plans such as permanent diff cycling, as the flow is brought more in line with how TF operates the provider e.g. the expected behavior.

### CODE REVIEW

File: provider/provider_yaml_test.go [hunk 1]
```diff
@@ -371,13 +371,13 @@ resources:
 		})
 
 		res := pulumiTest.Preview()
-		fmt.Printf("stdout: %s \n", res.StdOut)
-		fmt.Printf("stderr: %s \n", res.StdErr)
+		t.Logf("stdout: %s \n", res.StdOut)
```
Comment (t0yv0) [inline provider/provider_yaml_test.go:374]: Unrelated, but it helps to see the log messages in CI under the right sub-test.

File: patches/0066-Legacy-bucket-Read-now-sets-acl-and-force_destroy-de.patch [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
++	if acl, ok := d.GetOk("acl"); ok && acl.(string) != bucketACLDefaultValue {
+ 		if err := d.Set("grant", nil); err != nil {
+ 			return diag.Errorf("error resetting grant %s", err)
+ 		}
+@@ -1403,6 +1407,18 @@ func resourceBucketLegacyRead(ctx context.Context, d *schema.ResourceData, meta
+ 	}.String()
+ 	d.Set("arn", arn)
+ 
++	// Set ACL default if unset, this fixes resource import option operation under Pulumi.
++	_, gotGrant := d.GetOk("grant")
```
Comment (t0yv0) [inline patches/0066-Legacy-bucket-Read-now-sets-acl-and-force_destroy-de.patch:45]: Fixing import option for Legacy Bucket. Read now applies defaults.

File: examples/examples_yaml_test.go [hunk 3]
```diff
@@ -790,20 +654,15 @@ func TestMigrateRdsInstance(t *testing.T) {
...
-	      "hasDetailedDiff": true,
-              "detailedDiff": "*"
+	      "hasDetailedDiff": true
 	    }
 	  }
 	]`
 
 	t.Run("case1", func(t *testing.T) { replay(t, case1) })
 	t.Run("case2", func(t *testing.T) { replay(t, case2) })
-	t.Setenv("PULUMI_ENABLE_PLAN_RESOURCE_CHANGE", "true")
```
Comment (t0yv0) [inline examples/examples_yaml_test.go:804]: Removing cases not relevant anymore.

File: examples/examples_yaml_test.go [hunk 4]
```diff
@@ -377,12 +377,12 @@ outputs:
 // This replicates the diff when running `pulumi preview` on a aws.rds.Instance with
 // pulumi-aws v6.0.0 and state from pulumi-aws 5.42.0.
 //
-// The first test ensures we don't regress on https://github.com/pulumi/pulumi-aws/issues/2682
+// The first test ensures we don't regress on [[https://github.com/pulumi/pulumi-aws/issues/2682][pulumi/pulumi-aws#2682: RDS Instance replacement when upgrading to 6.0]]
```
Comment (t0yv0) [inline examples/examples_yaml_test.go:380]: Accidental change, let me revert.

File: provider/testdata/recorded/TestProviderUpgrade/TestOpenZfsFileSystemUpgrade/new-version/plan.json [hunk 5]
```diff
@@ -1,12 +1,12 @@
 {
+    "manifest": {
```
Comment (t0yv0) [inline provider/testdata/recorded/TestProviderUpgrade/TestOpenZfsFileSystemUpgrade/new-version/plan.json:2]: This got re-recorded. I think unnecessarily but including for now

**Comment by t0yv0** [comment]
@VenelinMartinov since eks Cluster is fixed via a special case, I'm deferring the EnableZeroDefaultSchemaVersion rollout for later.  I can go either way here, but my preference is to take one change at a time.

File: provider/provider_yaml_test.go [hunk 6]
```diff
@@ -548,7 +548,7 @@ type tagsTestStep struct {
 
 // TestAccDefaultTags tries to test all the scenarios that might affect provider defaultTags / resource tags
 // i.e. up, refresh, preview, import, etc
-func TestAccDefaultTags(t *testing.T) {
+func TestAccDefaultTagsWithImport(t *testing.T) {
```
Comment (t0yv0) [inline provider/provider_yaml_test.go:551]: There is another test called TestAccDefaultTags in the repo and I was getting confused.

File: patches/0066-Legacy-bucket-Read-now-sets-acl-and-force_destroy-de.patch [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
++	_, gotACL := d.GetOk("acl")
++	if !gotGrant && !gotACL {
++		d.Set("acl", bucketACLDefaultValue)
++	}
++
++	// Similarly, set force_destroy to default value if unset, to fix resource import operation.
++	if _, fdSet := d.GetOk("force_destroy"); !fdSet {
++		d.Set("force_destroy", false)
```
Comment (t0yv0) [inline patches/0066-Legacy-bucket-Read-now-sets-acl-and-force_destroy-de.patch:53]: Appreciate eyes on this one; other scenarios this affects are refresh, import proper (not through  resource option), and Bucket.get. I think it should be safe but can dig deeper if there are concerns.

**Review by VenelinMartinov — APPROVED** [review]
🚢

**Pull request merged**